netbox/utilities/socks.py

```diff
@@ -0,0 +1,76 @@
+from urllib.parse import urlparse
+from urllib3 import PoolManager, HTTPConnectionPool, HTTPSConnectionPool
+from urllib3.connection import HTTPConnection, HTTPSConnection
+
+
+# These Proxy Methods are for handling SOCKS connection proxy
+class ProxyHTTPConnection(HTTPConnection):
+    use_rdns = False
+
+    def __init__(self, *args, **kwargs):
+        socks_options = kwargs.pop('_socks_options')
+        self._proxy_url = socks_options['proxy_url']
+        super().__init__(*args, **kwargs)
+
+    def _new_conn(self):
+        from python_socks.sync import Proxy
```

Comment:
We should catch & re-raise `ImportError` here to include a helpful error message.

Reply:
It's in the code that is executed by the background task queue so the error is never propagated to the front-end.  In the RQ log there is the message:
```
Fetching remote data failed (ModuleNotFoundError): No module named 'python_socks'
```
Which should be similar to dulwich and any other missing library so I think a different error message might be more confusing?

Reply:
We should log an error which hints as to the root cause; something like "Configuring an HTTP proxy using SOCKS requires the python_socks library. Check that it has been installed."